### PR TITLE
Private/mmeeks/pollclean

### PR DESCRIPTION
**Pull request by mmeeks**
* Resolves: #9833  refining PR #9916
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] I have run `make prettier-write` and formatted the code.
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: net/Socket.cpp [hunk 1]
```diff
@@ -1505,13 +1505,13 @@ bool StreamSocket::checkRemoval(std::chrono::steady_clock::time_point now)
         const auto durLast =
             std::chrono::duration_cast<std::chrono::milliseconds>(now - getLastSeenTime());
         /// TO Criteria: Violate maximum idle (_pollTimeout default 64s)
-        const bool isIDLE = _pollTimeout > std::chrono::microseconds::zero() &&
-                           durLast > _pollTimeout;
+        const bool isIdle = _pollTimeout > std::chrono::microseconds::zero() &&
```
Comment (Ashod) [inline net/Socket.cpp:1489]: Yes, more than a couple of capitalized letters and it fast becomes obnoxious.

**Review by Ashod — APPROVED** [review]
Thanks, @mmeeks. Looks great. Restores the original structure and logic. Much more readable too.

**Review by sgothel — APPROVED** [review]
With this re-org `virtual Socket::checkRemoval` can be removed and `StreamSocket::checkRemoval` a plain method.
I can add this on top.

As discussed, if `SigUtil::getTerminationFlag()` checkRemoval 
should indeed `closeConnection()` right away to be save,
as polling might be stopped as in TerminatingPoll or kitPoll().

Looks great.

**Comment by sgothel** [comment]
`Restores exception guards` is very good, as `checkRemoval` also called into `_socketHandler->checkTimeout`.
While the latter is not a custom user callback, its more save this way.

Kicked jenkins ..

**Pull request merged**